tor/validation/formatting_validation.py

```diff
@@ -42,6 +42,12 @@
 # ```
 FENCED_CODE_BLOCK_PATTERN = re.compile("```.*```", re.DOTALL)
 
+# Regex to recognize unescaped hashtags which may render as headers.
+# Example:
+#
+# #Hashtag
+UNESCAPED_HEADER_PATTERN = re.compile(r"([^\\]|^)#[^ ].*")

```

Comment:
I think we can make this Regex a bit smarter. For example, the bot would currently warn for something like this:

```
\#hashtag1 #hashtag2 #hashtag3
```

or for this:

```
*#hashtag1*
```

However, both examples would display correctly on Reddit.
To fix that, we can require a paragraph linebreak before the number sign. You can take the regex for that from the `PROPER_SEPARATOR_PATTERN`, it should look like this: `\n[ ]*\n[ ]{,3}` (You can have spaces on the empty line and up to three spaces before the heading).

In general, we prefer to have false negatives to having false postitives, the latter could be very annoying for experienced transcribers.

Reply:
Changed the Regex to test for the paragraph linebreaks (provided regex) or start of string (this shouldn't happen since there's a header)